src/main/java/com/pubnub/api/managers/ReconnectionManager.java

```diff
@@ -78,11 +82,15 @@ private void registerHeartbeatTimer() {
             public void run() {
                 callTime();
             }
-        }, getNextInterval() * MILLISECONDS);
+        }, getNextIntervalInMilliSeconds());
     }
 
-    int getNextInterval() {
-        int timerInterval = LINEAR_INTERVAL;
+    private boolean maxConnectionIsNotSetToInfinite() {
```

Comment:
having negation (`Not`) in the middle of the name here is confusing,
how about `isInfiniteReconnections` ? (and flip the `if` to `==`, and fix call sites)

Reply:
But then condition would change to:
 if (!maxConnectionIsSetToInfinite() && failedCalls >= maxConnectionRetries) {
 
For me it is more clear to have
 if (maxConnectionIsNotSetToInfinite() && failedCalls >= maxConnectionRetries) {
then
 if (!maxConnectionIsSetToInfinite() && failedCalls >= maxConnectionRetries) {

What do you think?

Reply:
For me `maxConnectionIsNotSetToInfinite` reads better. I created this function to describe this mysterious condition:
maxConnectionRetries != -1
I assumed the function will not be reused. In situation `!isNotSomething()` is needed I would refactor it.

Anyway I will change it to `maxConnectionIsSetToInfinite`